### ISSUE TITLE
use http_basic login

### ISSUE THREAD

**Issue by hlb-schmidt** [ldaptools/ldaptools-bundle#25]
how can we use http_basic for https://github.com/ldaptools/ldaptools-bundle/blob/master/Resources/doc/Save-LDAP-Users-to-the-Database-After-Login.md ?

for example like http://symfony.com/doc/current/security/ldap.html#configuration-example-for-http-basic can

**Comment by ChadSikorra** [comment]
I think you could modify the firewall to be:

```yaml
security:

    # https://symfony.com/doc/current/security.html#b-configuring-how-users-are-loaded
    providers:
        ldap:
            id: ldap_tools.security.user.ldap_user_provider

    firewalls:
        # disables authentication for assets and the profiler, adapt it according to your needs
        dev:
            pattern: ^/(_(profiler|wdt)|css|images|js)/
            security: false

        main:
            http_basic:
                provider: ldap
            guard:
                authenticators:
                    - ldap_tools.security.ldap_guard_authenticator
```

Though I have not tested that yet.

**Comment by hlb-schmidt** [comment]
with above, no http_basic dialog popup and symfony is still trying to redirect to `/login`.

**Comment by ChadSikorra** [comment]
Hmmmm... ok. I'll have to take a closer look at how that's supposed to be implemented and add it as a feature. I've never wanted to do http_basic so I never thought to add the functionality initially.

**Comment by ChadSikorra** [comment]
I added HTTP Basic Auth support with this commit: https://github.com/ldaptools/ldaptools-bundle/commit/80c0e891d3a39266ea3ec25d389f68a0497fce85. I tested it and it seems to work fine for me. Let me know how it works.  I added some basic docs for configuring it here:

https://github.com/ldaptools/ldaptools-bundle/blob/master/Resources/doc/LDAP-Authentication-Provider.md#http-basic-authentication

Also updated the configuration reference for it.
